website/server/models/challenge.js

```diff
@@ -66,27 +66,33 @@ schema.statics.sanitizeUpdate = function sanitizeUpdate (updateObj) {
   return this.sanitize(updateObj, noUpdate);
 };
 
+// Returns true if user is the leader/owner of the challenge
+schema.methods.isLeader = function isChallengeLeader (user) {
+  return this.leader === user._id;
+};
+
 // Returns true if user is a member of the challenge
 schema.methods.isMember = function isChallengeMember (user) {
   return user.challenges.indexOf(this._id) !== -1;
 };
 
 // Returns true if the user can modify (close, selectWinner, ...) the challenge
 schema.methods.canModify = function canModifyChallenge (user) {
-  return user.contributor.admin || this.leader === user._id;
+  return user.contributor.admin || this.isLeader(user);
 };
 
-// Returns true if user has access to the challenge (can join)
-schema.methods.hasAccess = function hasAccessToChallenge (user, group) {
+// Returns true if user can join the challenge
+schema.methods.canJoin = function canJoinChallenge (user, group) {
   if (group.type === 'guild' && group.privacy === 'public') return true;
+  if (this.isLeader(user)) return true; // for when leader has left private group that contains the challenge
   return user.getGroups().indexOf(this.group) !== -1;
 };
 
 // Returns true if user can view the challenge
-// Different from hasAccess because you can see challenges of groups you've been removed from if you're partecipating in them
+// Different from canJoin because you can see challenges of groups you've been removed from if you're participating in them
 schema.methods.canView = function canViewChallenge (user, group) {
-  if (this.isMember(user)) return true;
-  return this.hasAccess(user, group);
+  if (this.isLeader(user) || this.isMember(user)) return true;
```

Comment:
the this.isLeader is also checked in canJoin, is it necessary to have it here as well?

Reply:
Yes, canView and canJoin are used at different times so they both need the check. canJoin is relevant only when the challenge owner is trying to rejoin their own challenge, not when they're trying to administer it without joining. And this has reminded me that I meant to respond to THI's post about that in the PR's comments.  I'll post a reply in a few minutes.

Reply:
After a DM from paglias, I looked more closely and canView calls canJoin so the isLeader check here is indeed unnecessary. I've removed it. :)